cpp/src/gandiva/bitmap_accumulator.cc

```diff
@@ -50,29 +65,20 @@ void BitMapAccumulator::IntersectBitMaps(uint8_t* dst_map,
 
     case 1: {
       // one src_maps_ bitmap. copy to dst_map
-      memcpy(dst_map, src_maps[0], num_bytes);
-      break;
-    }
-
-    case 2: {
-      // two src_maps bitmaps. do 64-bit ANDs
-      uint64_t* src_maps0_64 = reinterpret_cast<uint64_t*>(src_maps[0]);
-      uint64_t* src_maps1_64 = reinterpret_cast<uint64_t*>(src_maps[1]);
-      for (int64_t i = 0; i < num_words; ++i) {
-        dst_map64[i] = src_maps0_64[i] & src_maps1_64[i];
-      }
+      arrow::internal::CopyBitmap(src_maps[0], src_map_offsets[0], num_records, dst_map,
+                                  0);
       break;
     }
 
     default: {
-      /* > 2 src_maps bitmaps. do 64-bit ANDs */
-      uint64_t* src_maps0_64 = reinterpret_cast<uint64_t*>(src_maps[0]);
-      memcpy(dst_map64, src_maps0_64, num_bytes);
-      for (int64_t m = 1; m < nmaps; ++m) {
-        for (int64_t i = 0; i < num_words; ++i) {
-          uint64_t* src_mapsm_64 = reinterpret_cast<uint64_t*>(src_maps[m]);
-          dst_map64[i] &= src_mapsm_64[i];
-        }
+      // initialize all output bits to 0 (as required by BitmapAnd)
+      arrow::BitUtil::SetBitsTo(dst_map, 0, num_records, false);
```

Comment:
if the zerofilling is no longer required (after your change to bit_util.cc), can we drop this line ?

Reply:
yeah, I will remove this.